src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -98,26 +112,76 @@ static std::string machine_info()
   return model;
 }
 
-static std::vector<std::weak_ptr<xrt_core::device_linux>> mgmtpf_devices(16); // fix size
-static std::vector<std::weak_ptr<xrt_core::device_linux>> userpf_devices(16); // fix size
-static std::map<xrt_core::device::handle_type, std::weak_ptr<xrt_core::device_linux>> userpf_device_map;
-
 }
 
 namespace xrt_core {
 
+void
+system_linux::
+register_driver(std::shared_ptr<pci::drv> driver)
+{
+  if (driver->is_user())
+    driver->scan_devices(user_ready_list, user_nonready_list);
+  else
+    driver->scan_devices(mgmt_ready_list, mgmt_nonready_list);
+  driver_list.emplace_back(std::move(driver));
```

Comment:
Format: Insert new line
```suggestion

    driver_list.emplace_back(std::move(driver));
```

Reply:
sure.